src/lib.rs

```diff
@@ -9,3 +13,55 @@ pub mod good_fences_runner;
 pub mod import_resolver;
 mod path_utils;
 pub mod walk_dirs;
+
+#[napi]
+pub fn good_fences(
+    paths: Vec<String>,
+    project: String,
+    base_url: Option<String>,
+    root: String,
+    output: Option<String>,
+) {
+    let args = cli::Cli {
+        paths,
+        project,
+        base_url,
+        root,
+        output: output.unwrap_or("good-fences-violations.json".to_owned()),
+    };
+    run_evaluations(args);
+}
+
+pub fn run_evaluations(args: cli::Cli) {
+    let start = Instant::now();
+    let root = Path::new(args.root.as_str());
+    let tsconfig_path = args.project;
+    let mut tsconfig = import_resolver::TsconfigPathsJson::from_path(tsconfig_path).unwrap();
+
+    if args.base_url.is_some() {
+        tsconfig.compiler_options.base_url = args.base_url;
+    }
+
+    assert!(std::env::set_current_dir(&root).is_ok());
```

Comment:
This runs within the node process. Nothing on the api code path should change the working directory since that's not expected behaviour from the consumer.

Reply:
I think this change implies a significant change specially when it comes to evaluate the fences path, can we create a ticket to address this after integrating it with owa-builld/client-web?